docs/source/en/features/lazy_init.md

```diff
@@ -0,0 +1,76 @@
+# Lazy initialization
+
+Author: [Hongxiu Liu](https://github.com/ver217)
+
+**Prerequisite:**
+- [Train with booster](../basics/booster_api.md)
+
```

Comment:
Maybe we can put related paper here, for example this one: https://arxiv.org/abs/2102.13267

Reply:
Our implementation does not follow this paper.